vault/core.go

```diff
@@ -592,23 +592,6 @@ func (c *Core) handleRequest(req *logical.Request) (retResp *logical.Response, r
 			return nil, auth, ErrInternalError
 		}
 
-		sysView := c.router.MatchingSystemView(req.Path)
```

Comment:
This exact logic is in the token store, and this block is only executed if the auth is being returned from the token store. Having this logic here as well causes problems with e.g. periodic TTLs that are allowed to be greater than the mount max (for non-periodic).

See https://github.com/hashicorp/vault/blob/master/vault/token_store.go#L680